<launch>
  <param name="use_sim_time" value="true" />
  <arg name="id" default=""/>
  <arg name="client_number" default="2"/>
  <arg name="robot_namespace" default="firefly"/>

  <!-- User supplied arguments -->
  <arg name="show_rviz" default="false" doc="Automatically launch Rviz." />
  <arg name="scene_name" default="cvg_lab" />
  <arg name="sensor_name" default="firefly" />
  <arg name="visualize" default="false" />
  <arg name="verbose" default="true"/>
  <arg name="debug" default="false" />
  <arg name="pointcloud_topic" doc="ROS topic for pointcloud input" default="/firefly_$(arg id)/vi_sensor/camera_depth/depth/points"/>

  <group>
    <arg unless="$(arg debug)" name="launch_prefix" value=" " />
    <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />

    <node name="label_tsdf_client_$(arg id)" pkg="maskgraph" type="label_tsdf_server" output="log" args="-alsologtostderr" clear_params="true" launch-prefix="$(arg launch_prefix)">
      <remap from="pointcloud" to="$(arg pointcloud_topic)"/>
      <rosparam file="$(find maskgraph)/config/segment_map.yaml"/>
      <rosparam file="$(find maskgraph)/config/$(arg scene_name).yaml"/>
      <rosparam file="$(find coxgraph)/config/tsdf_server_default.yaml"/>
      <param name="meshing/visualize" value="$(arg visualize)" />
      <param name="enable_segment_map" value="false" />
      <param name="enable_local_segment_map" value="false" />
      <param name="verbose" value="$(arg verbose)" />
      <param name="publish_tsdf_map" value="true" />
      <param name="publish_esdf_map" value="true" />
      <param name="publish_slices" value="true" />
      <param name="publish_pointclouds" value="true" />
      <param name="publish_traversable" value="true"/>
      <param name="update_esdf_every_n_sec " value="0.5"/>
      <param name="publish_active_tsdf_every_n_sec " value="0.5"/>
      <param name="world_frame" value="map_$(arg id)"/>
      <param name="client_number" value="$(arg client_number)"/>
      <param name="client_id" value="$(arg id)"/>
      <param name="robot_namespace" value="$(arg robot_namespace)"/>
      <param name="baselink_frame" value="base_link"/>
      <param name="collision_radius" value="10.0"/>
    </node>

  </group>

</launch>
